### PR TITLE
add SERVER_MORE_RESULTS_EXISTS judgment

### PR DESCRIPTION
**Pull request by CJKkkk-315**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #https://github.com/matrixorigin/MO-Cloud/issues/3107

## What this PR does / why we need it:

补全响应client时的SERVER_MORE_RESULTS_EXISTS 设置

___

### **PR Type**
Bug fix

___

### **Description**
- Added a check for `SERVER_MORE_RESULTS_EXISTS` status flag in `respStreamResultRow` and `respMixedResultRow` functions.
- Updated calls to `WriteEOFOrOK` to use the new status variable instead of calling `ses.getStatusAfterTxnIsEnded(execCtx.reqCtx)` directly.
- Ensured the `SERVER_MORE_RESULTS_EXISTS` flag is set correctly based on whether the statement is the last one in the execution context.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>result_row_stmt.go</strong><dd><code>Add SERVER_MORE_RESULTS_EXISTS status flag handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
pkg/frontend/result_row_stmt.go

<li>Added a check for <code>SERVER_MORE_RESULTS_EXISTS</code> status flag.<br> <li> Updated calls to <code>WriteEOFOrOK</code> to use the new status variable.<br> <li> Ensured status flag is set correctly based on whether the statement is <br>the last one.<br>

</details>
    

  </td>
  <td><a href="https://github.com/matrixorigin/matrixone/pull/16712/files#diff-f4b9fba3825fe81e1de50097face2d704cffacd051f87aefa801d6a96c5aa675">+14/-4</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: pkg/frontend/result_row_stmt.go [hunk 1]
```diff
@@ -206,13 +206,17 @@ func (resper *MysqlResp) respStreamResultRow(ses *Session,
 	if execCtx.skipRespClient {
 		return nil
 	}
+	status := ses.getStatusAfterTxnIsEnded(execCtx.reqCtx)
```
Comment (daviszhen) [inline pkg/frontend/result_row_stmt.go:209]: 建议这段逻辑封装成单独的函数xxx。然后调用这个函数xxx。

Comment (daviszhen) [reply pkg/frontend/result_row_stmt.go:209]: 将修改也提交到1.2-dev上。

**Review by daviszhen — CHANGES_REQUESTED** [review]

File: pkg/frontend/result_row_stmt.go (hunk 1 above)
Comment (CJKkkk-315) [reply pkg/frontend/result_row_stmt.go]: isLastStmt判断直接添加到getStatusAfterTxnIsEnded函数中了

File: pkg/frontend/session.go [hunk 2]
```diff
@@ -1453,7 +1453,10 @@ func (ses *Session) StatusSession() *status.Session {
 // getStatusAfterTxnIsEnded
 // !!! only used after the txn is ended.
 // it may be called in the active txn. so, we
-func (ses *Session) getStatusAfterTxnIsEnded(ctx context.Context) uint16 {
+func (ses *Session) getStatusAfterTxnIsEnded(isLastStmt bool) uint16 {
+	if !isLastStmt {
+		return extendStatus(ses.GetTxnHandler().GetServerStatus()) | SERVER_MORE_RESULTS_EXISTS
```
[the remark this answers is not in the document]
Comment (CJKkkk-315) [reply pkg/frontend/session.go]: 修改完毕。

**Review by daviszhen — APPROVED** [review]

**Pull request merged**